### DISCUSSION TITLE
Starting development / IDE

### DISCUSSION THREAD

**Discussion by mLienhard** [aredden/electron-G14Control#38] [General]
Hi,
I'm very intereseted in this project due to i bought a G14 and there are not many possibilities to control the fan by myself (and this application seems to be able to do this).
Unfortunately my Windows is a german Windows and I'd like to keep this! ;-)

At the moment I've some issues with applying plans (which seems to work only sometimes) and the some small things (for example the temperature is read out correctly in the GUI)

I'm already familar with some C# coding but unfortunately I did not do anything with TypeScript (nor Python)
I tried to get the source code and created a project in Visual Studio but I stuck quite early in get it running.

Do you use an IDE at all for your development? If yes - which one? and can you also share all project related files?

another question: (I hardly dare to ask) would it be possible to transform the code to c#?

I really appreciate your work and would like to help a bit - but at the moment I'm not sure whether I'm able to...

**Comment by aredden** [comment]
I used vscode, since it's made by microsoft, and microsoft is also the original creator of typescript, so they go hand in hand. The one thing you will need is to add a file named `.env` with paths to certain files. It should be placed inside the /electron directory. These are (possibly not updated), but described in the README.md. I'll show what mine looks like right now (but altered):
```cmd
ATRO_LOC=C:/path/to/cloned/repo/electron/atrofac-cli/atrofac-cli.exe
RADJ_LOC=C:/path/to/cloned/repo/electron/ryzenadj/ryzenadj.exe
CONFIG_LOC=C:/path/to/cloned/repo/electron/src/config/config.dev.json
SCREEN_REF_LOC=C:/path/to/cloned/repo/electron/screen-resolution/ChangeScreenResolution.exe
RESTART_GPU_LOC=C:/path/to/cloned/repo/electron/restartgpu/RestartGPU.exe
RMOB_LOC=C:/path/to/cloned/repo/electron/renoir-mobile/renoir-mobile.exe
```
You would obviously replace C:/path/to/cloned/repo with your paths. Those are the helper programs that are packaged with the application.

To run the application in development, you would need to run the command `npm start` in the main repository directory.. and then `npm run watch` in the /electron directory which is one step below the main repo directory.

The things you would need to change are probably located in.. `electron\src\IPCEvents\TempPerf\Perfmon.ts` for the battery and cpu temperature performance counters. You can find your performance counters in cmd by running "typeperf -q". You would just need to find the translated names of the counters. Also, for powercfg boost and graphics stuff, you would need to go into `electron\src\IPCEvents\CPUBoostListeners.ts` and `electron\src\IPCEvents\powercfg\Powercfg.ts` and probably also `electron\src\IPCEvents\gpu\DiscreteGPU.ts` which is mostly regex parsing of text responses from the powercfg command, and executing powercfg commands.

I hope that explains it :)

Also, translating it to C# *would* be possible, but a massive amount of work. I do know that there is an electron library for C#, though I haven't used it.

**Reply by mLienhard** [reply]
Hi,
thanks for your additional help (although most of your comments were alread in the readme (sorry)
to get the npm isntaller properly running i needed an additional command "npm i -g windows-build-tools" to get the toolchain running
EDIT: I just saw that this is also alreay writting in the readme... (reading the instructions is not one of my strength :-/ )

Now i'm able to run the tool with some additional outputs.
Next step: try to understand what is different in my german windows!

Thanks again!
